kv/etcd/store.go

```diff
@@ -188,16 +216,29 @@ func (c *client) Watch(key string) (kv.ValueWatch, error) {
 	return w, err
 }
 
-func (c *client) watchChan(key string) clientv3.WatchChan {
-	return c.watcher.Watch(
-		context.Background(),
-		c.opts.KeyFn()(key),
-		// periodically (appx every 10 mins) checks for the latest data
-		// with or without any update notification
-		clientv3.WithProgressNotify(),
-		// receive initial notification once the watch channel is created
-		clientv3.WithCreatedNotify(),
-	)
+func (c *client) watchChanWithTimeout(key string) (clientv3.WatchChan, error) {
+	doneCh := make(chan struct{})
+
+	var watchChan clientv3.WatchChan
+	go func() {
+		watchChan = c.watcher.Watch(
+			context.Background(),
```

Comment:
Can we directly pass in a context with timeout here instead of doing a select down below?

Also if it does time out, is there a way to clean up the watcher?